### PR TITLE
Add servers property to listener resource

### PR DESCRIPTION
**Pull request by Stromweld**
### Description

Allows user to specify servers for listeners for simple setups.

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license           'Apache-2.0'
 description       'Installs and configures haproxy'
 long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '6.2.3'
+version           '6.2.4'
 source_url        'https://github.com/sous-chefs/haproxy'
```
Comment (damacus) [inline metadata.rb]: Please revert this change.

We'll bump this as part of a separate change.

Comment (Stromweld) [reply metadata.rb]: sure no problem

**Comment by Stromweld** [comment]
This fixes #298

**Comment by damacus** [comment]
I've been thinking about this, any reason we can't use the backend resource for this? 

As that's what it's generally for!

**Comment by Stromweld** [comment]
That creates a separate backend. This allows for that but also allows for simple listeners that have their own servers instead of associating to a separate backend that is configured differently and shows up as separate parts in the stats pages.

**Comment by Stromweld** [comment]
I didn’t remove the current of being able to specify a separate backend.

**Comment by Stromweld** [comment]
@damacus @rshade is there anything else I can do to get this PR merged or answer any questions?

**Comment by rshade** [comment]
Let me work with @damacus and I will try and get this merged this week.

**Comment by Stromweld** [comment]
thanks, I hit him up on slack to discuss further but he was at end of day so I told him I'll try back tomorrow earlier in the day in the sous-chefs channel.
